### ISSUE TITLE
Use any copula from `Copulas.jl`

### ISSUE THREAD

**Issue by lrnv** [AnderGray/ProbabilityBoundsAnalysis.jl#42]
Hey, 

What would be needed for your code to use any copulas from [ `Copulas.jl`](https://github.com/lrnv/Copulas.jl)? It might reduce maintenability burden in your `copulas.jl` file, and allow for many more parametric (or non-parametric) copula assumptions, as the possibilities of `Copulas.jl` allow many more models that what you have here. 

If you are interested in making this move, I might try to sketch a PR, but I'll need your help ;)

It would also allow for multivariate pboxes.

**Comment by AnderGray** [comment]
Hi! 

Thanks for opening this. Yes, I think this would be beneficial. We originally had [BivariateCopulas.jl](https://github.com/AnderGray/BivariateCopulas.jl) as a dependence, but we decided to re-implement here as we were playing with some new ideas. I think moving to your more general package would be great.

I don't think it's very difficult, but there are a couple of features I'd like to keep. 

- We should allow for parametric [imprecise copulas](https://www.sciencedirect.com/science/article/pii/S0165011414004539), which are a set of copulas given by the point-wise upper and lower bounds (e.g., `[W, M]` for bounds on all copulas). We currently make these by introducing interval parameters to a copula family, and working out the bounds, which for the current 2-copulas that we have is just the envelope of interval endpoints. Not sure how this generalises to higher dimensions. For the first PR, we could leave this out for copulas higher than 2.

- We use an (experimental) outer approximation to copulas in the same way we do for p-boxes. P-boxes and distributions are represented using a finite vector of inverse cdfs evaluated on a grid in `[0,1]`. Bounds on all probabilistic quantities (samples, cdf, measure) are then derived in terms of this representation. We experimented with doing the same with copulas:
https://github.com/AnderGray/ProbabilityBoundsAnalysis.jl/blob/319712bd5a7bc58bbab546b6f311a4efab306419/src/pbox/copulas.jl#L50-L53 where `cdfU` and `cdfD` are the 'up' and 'down' values of the copula on the same probability grid as the marginals. This also allows for a nice representation of imprecise copulas. This should be quite easy though, here's how independence is defined https://github.com/AnderGray/ProbabilityBoundsAnalysis.jl/blob/319712bd5a7bc58bbab546b6f311a4efab306419/src/pbox/copulas.jl#L394-L402

- `C-` or `H-volume`: currently we only have the 2D version, which we call `mass` (for probability mass,  `measure` is more a exact name): https://github.com/AnderGray/ProbabilityBoundsAnalysis.jl/blob/319712bd5a7bc58bbab546b6f311a4efab306419/src/pbox/copulas.jl#L190 We actually have a N-dimensional Julia implementation of [this](https://github.com/AnderGray/Hvolume-Matlab) in which we haven't added yet. But maybe this should be added to `Copulas.jl` ? :)

In short, I think a copula type should be kept (maybe calling it something else), and move all the cdf and sampling calls to your package.

**Comment by lrnv** [comment]
I now implemented `Copulas.measure(C,u,v)`, which computes the C-volume of the hypercube [u,v] in any dimensions.

Furthermore, I think that the computation of the grid points should stay in `ProbabilityBoundsAnalysis.jl` for the first draft of this PR: only the generation of these points might be standardized to a call to any copula from `Copula.jl`, and thus the upper/lower part of the mechanism will stay here. 

Do you think the tests of this repo are enough to allow me to make these changes without breaking anything you need ? If you feel confortable abut this, i'll try to make a PR that passes your tests.
